Add support for HTTP PATCH method

atreya2011 opened the pull request:
I have created a separate PR for adding support for HTTP PATCH method. 🙇🏼

lyonlai commented:
Can you add a simple integration test for the patch method? @atreya2011 . I've add some instructions in the `integration_tests/README.md`.

atreya2011 commented:
@lyonlai I added a simple integration test for HTTP Patch and in the process, upgraded tools.go and other files to use the latest generators for Go, gRPC and gRPC gateway

https://pkg.go.dev/google.golang.org/protobuf
https://github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway
https://google.golang.org/grpc/cmd/protoc-gen-go-grpc
https://google.golang.org/protobuf/cmd/protoc-gen-go

lyonlai commented on go.mod:
```diff
@@ -3,20 +3,23 @@ module github.com/grpc-ecosystem/protoc-gen-grpc-gateway-ts
-	github.com/grpc-ecosystem/grpc-gateway v1.15.2
+	github.com/golang/protobuf v1.5.2
+	github.com/google/uuid v1.2.0 // indirect
+	github.com/grpc-ecosystem/grpc-gateway/v2 v2.3.0
```
Although it's good to know this generator is working with v2 version of `grpc-gateway`, could we please don't make a sudden major version change like this? For reasons we are still using `grpc-gateway` v1, and it will be for a while. 

The support for different `grpc-gateway` versions in test is something needs to think about. And should be left out for this PR.

atreya2011 replied:
I understand. My apologies, I will revert these changes 🙇🏼

lyonlai commented on integration_tests/service.go:
```diff
@@ -5,7 +5,9 @@ import (
 
-type RealCounterService struct{}
+type RealCounterService struct {
+	UnimplementedCounterServiceServer
```
not sure why do you need this here

atreya2011 replied:
golangci-lint was giving a warning, and hence I added it there, will remove it 🙇🏼

atreya2011 commented:
@lyonlai My apologies once again for the sudden major version change of `grpc-gateway`. I have reverted it to version 1 and implemented the tests. Let me know if there are any other changes required 🙏🏼

lyonlai commented:
that's alright. awesome. I'll have it merged now. thank you for your contribution @atreya2011 .

atreya2011 commented:
Thank you for merging this 🙇🏼
